src/key_extractor.rs

```diff
@@ -17,6 +16,14 @@ pub trait KeyExtractor: Clone {
     /// Extraction method
     fn extract(&self, req: &ServiceRequest) -> Result<Self::Key, Self::KeyExtractionError>;
 
+    /// Error function, will pass [`Self::KeyExtractionError`] to it to return the response error
+    /// when the [`Self::extract`] failed [Read more]
+    ///
+    /// [Read more]: https://docs.rs/actix-web/4.1.0/actix_web/error/index.html#functions
+    fn response_error(&self, err: &Self::KeyExtractionError) -> actix_web::Error {
```

Comment:
Is there a particular reason why this only gets a reference to the error? I think this should be an owned type instead.

Reply:
I said that it could be developed in the future, we need the error, so I passed a reference

Reply:
I'm sorry, I don't understand your reasoning. It's not a big deal of course, but when we can pass is as owned type we should to prevent unnecessary cloning. Also, I don't understand what would change in the future to prevent this.